task-2/src/model/user/model.ts

```diff
@@ -0,0 +1,106 @@
+import { v4 as uuidv4 } from "uuid";
+import {
+  getUserSchema,
+  getUserSchemaForCreation,
+  getUserSchemaForUpdate,
+  suggestQueryParamsSchema,
+} from "./schema";
+import type {
+  User,
+  UserSchema,
+  CreateUserData,
+  CreateUserSchema,
+  UpdateUserData,
+  UpdateUserSchema,
+} from "./schema";
+import z from "zod";
+import { users } from "./data";
+import {
+  getUniqueLoginErrorMessage,
+  getUsersDictionary,
+  getUsersLoginSet,
+  sortUsersByLogin,
+} from "./utils";
+
+class UserModel {
+  private readonly _list: User[];
+  private readonly _dict: Record<string, User>;
+  private readonly _logins: Set<string>;
+
+  constructor(data: User[]) {
+    this._list = data;
+
+    this._dict = getUsersDictionary(data);
+
+    this._logins = getUsersLoginSet(data);
+  }
+
+  get users() {
+    return this._list;
+  }
+
+  getById(id: string) {
+    const user = this._dict[id];
+
+    if (!user) {
```

Comment:
I think here should be check for isDeleted == true

Reply:
Yes, I completely forgot about the delete operation. Added